lib/balena-lint.ts

```diff
@@ -241,6 +247,11 @@ export const lint = async (passedParams: any) => {
 				'Treat input files as test sources to perform extra relevant checks',
 			type: 'boolean',
 		})
+		.option('t', {
+			describe:
+				'Path to a tsconfig.json file to enable lint rules that rely on type information',
+			type: 'string',
```

Comment:
See this PR's changes to `package.json` for an example of how the option is used. Another example is this draft `balena-cli` PR: https://github.com/balena-io/balena-cli/pull/2389/files